### PR TITLE
Make JSONForms a peer dependency

### LINKED ISSUE #60: Use JSON Forms as peer dependency
Currently, JSONForms is a hard dependency to a specific version.
We want to use it as a peer dependency to allow adopters choosing the version. E.g. this allows adopters to use the latest version without a new tree editor release.

### LINKED ISSUE #74: Update to React 18
Currently we depend on React 16 but Theia is already using React 18. We should thus update

### PR DESCRIPTION
**Pull request by eneufeld**
Fix #60

### CODE REVIEW

**Review by lucas-koehler — COMMENTED** [review]
Quick note: This will break the dependency badge in the root README because it extracts the jsonforms version from the package.json. You need to adapt this to point to the peerDependencies instead

**Comment by eneufeld** [comment]
I also updated React now. 

fixes #74

File: configs/base.eslintrc.json [hunk 1]
```diff
@@ -7,37 +8,28 @@
...
-      "pragma": "React",
-      "version": "detect",
-      "flowVersion": "0.53"
-    }
-  },
   "env": {
     "browser": true,
     "mocha": true,
-    "es2017": true
+    "es6": true
```
Comment (lucas-koehler) [inline configs/base.eslintrc.json:24]: why use es6 here when [configs/base.tsconfig.json](https://github.com/eclipse-emfcloud/theia-tree-editor/pull/73/files#diff-69540381ae31eb557dbe0862c952999d8efe4e4e3b5bc84cc1916f8e440686d6) uses es2017 as lib and target

**Review by lucas-koehler — APPROVED** [review]
Generally LGTM. I just have one minor question. Also the CI fails but AFAIK that was already the case before this PR. Locally the build worked fine.
**EDIT:** I saw the latest build on master worked so if the restarted build of this PR fails again it might need some investigation after all.

**Comment by ndoschek** [comment]
Regarding the build: The upgrade of Theia requires more resources for the build container as we get Javascript out of memory exceptions. See `emfcloud-modelserver-theia`'s[Jenkinsfile](https://github.com/eclipse-emfcloud/emfcloud-modelserver-theia/blob/master/Jenkinsfile) for example, where we doubled the resources.

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
CI fails while it works on master. Please check what could be the problem

File: configs/base.eslintrc.json (hunk 1 above)
Comment (eneufeld) [reply configs/base.eslintrc.json:24]: I copied it over from the `emfcloud-modelserver-theia` config . This is the feature set for eslint, there is no option to use es2017 features. es6 is the only offered option in that regard.

**Comment by eneufeld** [comment]
@ndoschek Thank you for pointing this out. I had copied the the Jenkins file already but missed to pull first :face_with_spiral_eyes:

File: .eslintrc.js [hunk 2]
```diff
@@ -1,17 +1,46 @@
...
+                'header/header': [
+                    2,
+                    'block',
+                    [
+                        {
+                            pattern: '[\n\r]+ \\* Copyright \\([cC]\\) \\d{4}(-\\d{4})? .*[\n\r]+',
+                            template: `********************************************************************************
+  * Copyright (c) ${year} EclipseSource and others.
+  *
+  *
```
Comment (ndoschek) [inline .eslintrc.js:32]: One new line is sufficient here

```suggestion
  *
```

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks Eugen, looks good to me overall! :+1: 

Just one minor comment inline and there is one ESLint warning left, would be nice to fix that as well:
```
@eclipse-emfcloud/theia-tree-editor: .../theia-tree-editor/theia-tree-editor/src/browser/tree-editor-contribution.ts
@eclipse-emfcloud/theia-tree-editor:   28:1  warning  Opening curly brace does not appear on the same line as controlling statement  brace-style
@eclipse-emfcloud/theia-tree-editor: ✖ 1 problem (0 errors, 1 warning)
```

In general, this repository suggests to use Node 12 (in README and in the main package.json engine `"node": ">=12.14.1"`).
With the supported Theia version, this is outdated as it expects version `"^14.15.0 || >=16.0.0"`. I suggest to switch to Node 16 already (also fine by me to do in a follow-up).

**Review by ndoschek — COMMENTED** [review]
Thank you for the update @eneufeld, overall it looks good, however we need to limit the Node version to 16 (see comment inline).
Also, I spotted an outdated section in the detail README of the theia-tree-editor.

**Review by ndoschek — APPROVED** [review]
Thank you for the update @eneufeld!

**Pull request merged**